nixos/modules/security/acme.nix

```diff
@@ -104,7 +104,12 @@ let
     mkHash = with builtins; val: substring 0 20 (hashString "sha256" val);
     certDir = mkHash hashData;
     domainHash = mkHash "${concatStringsSep " " extraDomains} ${data.domain}";
-    othersHash = mkHash "${toString acmeServer} ${data.keyType}";
+    othersHash = mkHash (
+      "${toString acmeServer} ${data.keyType}"
+      + (
+        optionalString (versionOlder "20.09" config.system.stateVersion) data.email
```

Comment:
This looks good. I just had a thought that might also apply in general in these stateVersion checks.

Should we, at some point, throw a warning that a user should migrate (at their pace within a release) so we can realistically remove backwards compatibility at some point?

Reply:
@m1cr0man any opinions on that?

Reply:
So in the newer PR, I actually done away with this logic. This was written under the assumption that there was still rate limit issues with multiple certs being generated, but that's not the case now.

Reply:
Oh also worth noting I added a note on this matter to the [release notes](https://github.com/NixOS/nixpkgs/pull/106857/files#diff-e5a39d8376b580ef7bb2eb1a17900ed609b638aa453fb04f05569bfff04a0173R263).

Reply:
So should this PR be changed before we merge it as requested in https://github.com/NixOS/nixpkgs/pull/101482#issuecomment-745576658? Or do you want to close this one and incorporate it in the other?

Reply:
I will make the necessary changes to this PR so it can be merged. I worry it might take too long for the other PR to be merged.